crypto/encode_decode/decoder_lib.c

```diff
@@ -733,6 +747,7 @@ static int decoder_process(const OSSL_PARAM params[], void *arg)
     struct decoder_process_data_st new_data;
     const char *data_type = NULL;
     const char *data_structure = NULL;
+    const char *start_input_type = ctx->start_input_type;
```

Comment:
So we save this here to restore it on return? 
A comment to that effect would be nice ...

Reply:
Yes, I have no specific evidence that the restore is actually needed, but it seemed the cautious thing to do.